aws-iot-topicrule/src/main/java/com/amazonaws/iot/topicrule/ClientBuilder.java

```diff
@@ -0,0 +1,24 @@
+package com.amazonaws.iot.topicrule;
+
+import software.amazon.awssdk.core.client.config.ClientOverrideConfiguration;
+import software.amazon.awssdk.core.retry.RetryPolicy;
+import software.amazon.awssdk.services.iot.IotClient;
+
+public class ClientBuilder {
+    private static volatile IotClient iotClient;
+
+    static IotClient getClient() {
+        if (iotClient != null) {
+            return iotClient;
+        }
+
+        synchronized (ClientBuilder.class) {
+            iotClient = IotClient.builder()
+                    .overrideConfiguration(ClientOverrideConfiguration.builder()
+                            .retryPolicy(RetryPolicy.builder().numRetries(3).build())
```

Comment:
lets put an adaptive retry policy here.

And I am presuming this pkg does not support Spring/Guice so we have to use this Singleton factory?